packages/vitest/src/node/reporters/base.ts

```diff
@@ -66,6 +67,32 @@ export abstract class BaseReporter implements Reporter {
     }
   }
 
+  onTestCaseResult(testCase: TestCase): void {
+    if (testCase.result().state === 'failed') {
+      this.logFailedTask(testCase.task)
+    }
+  }
+
+  onTestSuiteResult(testSuite: TestSuite): void {
+    if (testSuite.state() === 'failed') {
+      this.logFailedTask(testSuite.task)
+    }
+  }
+
+  onTestModuleEnd(testModule: TestModule): void {
+    if (testModule.state() === 'failed') {
+      this.logFailedTask(testModule.task)
+    }
+  }
+
+  private logFailedTask(task: Task) {
+    if (this.ctx.config.silent === 'passed-only') {
+      for (const log of task.logs || []) {
+        this.onUserConsoleLog(log, true)
```

Comment:
`force = true` skips `this.shouldLog` which checks custom `onConsoleLog` handler. I think we should still take this account for further filtering by the user?

https://github.com/vitest-dev/vitest/blob/e2c570b687316d636fc257a7011191667f0c765a/packages/vitest/src/node/reporters/base.ts#L340-L348

Reply:
Oh good point! The `force` was needed to bypass `if (this.ctx.config.silent)` here. 

I'll need to add test case that uses both `--silent` and `onConsoleLog`.